src/actions.rs

```diff
@@ -19,125 +20,16 @@ pub struct Actions {
 }
 
 fn set_movement_actions(mut actions: ResMut<Actions>, keyboard_input: Res<Input<KeyCode>>) {
-    if GameControl::Up.just_released(&keyboard_input)
-        || GameControl::Up.pressed(&keyboard_input)
-        || GameControl::Left.just_released(&keyboard_input)
-        || GameControl::Left.pressed(&keyboard_input)
-        || GameControl::Down.just_released(&keyboard_input)
-        || GameControl::Down.pressed(&keyboard_input)
-        || GameControl::Right.just_released(&keyboard_input)
-        || GameControl::Right.pressed(&keyboard_input)
-    {
-        let mut player_movement = Vec2::ZERO;
-
-        if GameControl::Up.just_released(&keyboard_input)
-            || GameControl::Down.just_released(&keyboard_input)
-        {
-            if GameControl::Up.pressed(&keyboard_input) {
-                player_movement.y = 1.;
-            } else if GameControl::Down.pressed(&keyboard_input) {
-                player_movement.y = -1.;
-            } else {
-                player_movement.y = 0.;
-            }
-        } else if GameControl::Up.just_pressed(&keyboard_input) {
-            player_movement.y = 1.;
-        } else if GameControl::Down.just_pressed(&keyboard_input) {
-            player_movement.y = -1.;
-        } else {
-            player_movement.y = actions.player_movement.unwrap_or(Vec2::ZERO).y;
-        }
-
-        if GameControl::Right.just_released(&keyboard_input)
-            || GameControl::Left.just_released(&keyboard_input)
-        {
-            if GameControl::Right.pressed(&keyboard_input) {
-                player_movement.x = 1.;
-            } else if GameControl::Left.pressed(&keyboard_input) {
-                player_movement.x = -1.;
-            } else {
-                player_movement.x = 0.;
-            }
-        } else if GameControl::Right.just_pressed(&keyboard_input) {
-            player_movement.x = 1.;
-        } else if GameControl::Left.just_pressed(&keyboard_input) {
-            player_movement.x = -1.;
-        } else {
-            player_movement.x = actions.player_movement.unwrap_or(Vec2::ZERO).x;
-        }
-
-        if player_movement != Vec2::ZERO {
-            player_movement = player_movement.normalize();
-            actions.player_movement = Some(player_movement);
-        }
+    let player_movement = Vec2::new(
+        is_pressed(GameControl::Right, &keyboard_input)
+            - is_pressed(GameControl::Left, &keyboard_input),
+        is_pressed(GameControl::Up, &keyboard_input)
+            - is_pressed(GameControl::Down, &keyboard_input),
+    );
+
+    if player_movement != Vec2::ZERO {
+        actions.player_movement = Some(player_movement);
```

Comment:
We still need to normalize here, otherwise orthogonal movement is too fast.

Reply:
Added the normalization back ✅